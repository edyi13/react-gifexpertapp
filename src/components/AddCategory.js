import React, { useState } from 'react';
import PropTypes from 'prop-types';

//Se llama setCategories del componente GifExpertApp
//Se pasa setCategories como props desestructuradas (propiedades) 
export const AddCategory = ( {setCategories} ) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        //console.log(e.target.value)
        setInputValue(e.target.value);
    };

    const hanldeSubmit = (e) => {
        e.preventDefault();

        if( inputValue.trim().length > 2 ){
            setCategories(cats => [ inputValue, ...cats ]);
            setInputValue('');
        }
    };

    return (
            <form onSubmit={ hanldeSubmit }>
                <input
                    type="text"
                    value={ inputValue }
                    onChange= { handleInputChange  }/>
            </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
