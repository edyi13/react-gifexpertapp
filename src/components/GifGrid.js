import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import React, { useState, useEffect } from 'react'
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( {category} ) => {

    // const [count, setCount] = useState(0);
    // const [images, setImages] = useState([])
    const { data:images, loading} = useFetchGifs(category);
    // useEffect(() => {
    //     //se usa de esta forma por ser una promesa
    //     getGifs(category).then(imgs => setImages(imgs));
    // },[ category ]) // se envia si la categoria cambia para ejecutar el efecto

    
    
    return (
        <>
        <h3 className='animate__animated animate__bounce animate__fadeIn'>{ category } </h3>

        { loading && <p className='animate__animated animate__bounce animate__flash'>Loading</p> }
        
        <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem
                            key = {img.id}
                            //uso del operado spread ---> ...
                            {...img}
                        />
                    ))
                }
        </div>
        </>
        
    )
}
