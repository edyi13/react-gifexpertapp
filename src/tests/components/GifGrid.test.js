import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid/>', () => {
    const category = 'Monkey D. Luffy';
    // let wrapper = shallow(<GifGrid category={ category } />);  

    // beforeEach( () => {
    //     wrapper = shallow(<GifGrid category={ category } />);  
    // });  

    test('Deberia de mostrase <GifGrid/> correctamente', () => {
        //Arrange
        //Act
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={ category } />);  

        //Assert
        expect(wrapper).toMatchSnapshot();
    })

    test('Deberia de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        //Arrange
        const gifs = [{
            id:'ABC',
            url: 'http://gph.is/2tsV7lY',
            title: 'Super Vegeta'
        },
        {
            id:'DEF',
            url: 'http://gph.is/2tsV7lY',
            title: 'Super Vegeta'
        }];
        //Act
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={ category } />);  

        //Asert
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    
    
});
