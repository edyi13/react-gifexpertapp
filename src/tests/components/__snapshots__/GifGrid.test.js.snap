// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid/> Deberia de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce animate__fadeIn"
  >
    Monkey D. Luffy
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Super Vegeta"
      url="http://gph.is/2tsV7lY"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Super Vegeta"
      url="http://gph.is/2tsV7lY"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid/> Deberia de mostrase <GifGrid/> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce animate__fadeIn"
  >
    Monkey D. Luffy
     
  </h3>
  <p
    className="animate__animated animate__bounce animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
